### PR TITLE
add duration to recordings list

### PR DESCRIPTION
**Pull request by narthur**

### CODE REVIEW

**Comment by narthur** [comment]
https://audioverse-next-git-sermon-list-speakers.avorg.vercel.app/en/sermons/page/1

File: src/components/molecules/recordingListEntry.tsx [hunk 1]
```diff
@@ -4,6 +4,22 @@ import React from 'react';
 
 import styles from '@components/molecules/recordingListEntry.module.scss';
 
+const formatDuration = (seconds: number, m = 0, h = 0): string => {
+	seconds = Math.round(seconds);
+
+	if (seconds >= 60 * 60) {
+		return formatDuration(seconds - 60 * 60, m, h + 1);
```
Comment (mattleff) [inline src/components/molecules/recordingListEntry.tsx:11]: I'm not a fan of the recursive solution here. Calling `formatDuration(3554)` recursively calls `formatDuration()` 60 times. How about something like this (haven't tested this):

```ts
const formatDuration = (duration: number): string => {
     duration = Math.round(duration);
     const hours = Math.floor(duration / 3600);
     const minutes = Math.floor((duration - (hours * 3600)) / 60);
     const seconds = String(duration % 60).padStart(2, '0');
     return `${hours ? hours + ':' + String(minutes).padStart(2, '0') : minutes}:${seconds}`;
 };
```

**Review by mattleff — APPROVED** [review]
Looks good other than one comment 🏎️

File: src/components/molecules/recordingListEntry.tsx (hunk 1 above)
Comment (narthur) [reply src/components/molecules/recordingListEntry.tsx:11]: Nice, works great!

**Pull request merged**